src/main/java/com/simpligility/maven/plugins/android/phase08preparepackage/DexMojo.java

```diff
@@ -570,32 +595,70 @@ private static File getJavaExecutable()
         return new File( javaHome + slash + "bin" + slash + "java" );
     }
 
-    private File generateMainDexClassesFile() throws MojoExecutionException 
+    private File generateMainDexClassesJar( CommandExecutor executor ) throws MojoExecutionException 
     {
-        CommandExecutor executor = CommandExecutor.Factory.createDefaultCommmandExecutor();
-        executor.setLogger( getLog() );
-        List< String> commands = new ArrayList< String>();
-        commands.add( "--output" );
+        List< String> commands = jarDefaultCommands();
+        commands.add( getAndroidSdk().getProguardJarPath() );
+        commands.add( "-dontnote" );
+        commands.add( "-dontwarn" );
+        commands.add( "-forceprocessing" );
+        commands.add( "-dontoptimize" );
+        commands.add( "-dontpreverify" );
+        commands.add( "-dontobfuscate" );
+        
+        Set< File> inputFiles = getDexInputFiles();
+        for ( File inputFile : inputFiles ) 
+        {
+            commands.add( "-injars" );
+            commands.add( inputFile.getAbsolutePath() + "(!META-INF/**)" );     
+        } 
+        
+        commands.add( "-libraryjars" );
+        commands.add( getAndroidSdk().getShrinkedAndroidJarPath() );
+        
+        commands.add( "-include" );
+        commands.add( getAndroidSdk().getMainDexClassesRulesPath() );
+        
+        commands.add( "-outjars" );
+        File mainDexClassesJar = new File( targetDirectory, "mainDexClasses.jar" );
+        commands.add( mainDexClassesJar.getAbsolutePath() );
+        
+        getLog().info( "Generating main dex classes jar : " + mainDexClassesJar );
+        executeJava( commands, executor );
+        
+        return mainDexClassesJar;
+    }
+    
+    private File generateMainDexClassesList( CommandExecutor executor ) throws MojoExecutionException 
+    {
+        File mainDexClassesJar = generateMainDexClassesJar( executor );
+        List< String> commands = javaDefaultCommands();
         
-        File mainDexClasses = new File( targetDirectory, "mainDexClasses.txt" );
-        commands.add( mainDexClasses.getAbsolutePath() );
+        commands.add( "-Djava.ext.dirs=" + getAndroidSdk().getBuildToolsLibDirectoryPath() );
+        commands.add( "com.android.multidex.MainDexListBuilder" );        
+        commands.add( mainDexClassesJar.getAbsolutePath() );
         
         Set< File> inputFiles = getDexInputFiles();  
         StringBuilder sb = new StringBuilder();
-        sb.append( '"' ).append( StringUtils.join( inputFiles, File.pathSeparatorChar ) ).append( '"' );
```

Comment:
From what I remember this was necessary for things to work on OSX, Linux and windows..